type-tests/files/createSlice.typetest.ts

```diff
@@ -147,6 +147,25 @@ function expectType<T>(t: T) {
   expectType<string>(counter.actions.concatMetaStrLen('test').meta)
 }
 
+/*
+ * Test: returned case reducer has the correct type
+ */
+{
+  const counter = createSlice({
+    name: 'counter',
+    initialState: 0,
+    reducers: {
+      increment(state, action: PayloadAction<number>) {
+        return state + action.payload
+      }
+    }
+  })
+
+  expectType<(state: number, action: PayloadAction<number>) => number | void>(
+    counter.caseReducers.increment
+  )
```

Comment:
these two additional tests should be failing, but currently don't

```typescript
  // typings:expect-error
  expectType<(state: number, action: PayloadAction<string>) => number | void>(
    counter.caseReducers.increment
  )

  // typings:expect-error
  expectType<(state: number, action: PayloadAction<string>) => number | void>(
    counter.caseReducers.someThingNonExistant
  )
```

Reply:
Thanks!  These were a _huge_ help!

Reply:
I'd add one non-error-check for `decrement` here, as currently you're just testing for expect-errors on that one.

Reply:
something like this one.
 ```typescript
  // Should match positively for reducers with prepare callback
  expectType<(state: number, action: PayloadAction<number>) => number | void>(
    counter.caseReducers.decrement
  )
 ```